src/main/java/net/dv8tion/jda/core/requests/restaction/ChannelAction.java

```diff
@@ -0,0 +1,178 @@
+    {
+        Args.notNull(role, "Override Role");
+        if (!role.getGuild().equals(guild))
+            throw new IllegalArgumentException("Specified Role is not in the same Guild!");
+
+        String id = role.getId();
+        PromisePermissionOverride override = new PromisePermissionOverride(ROLE_TYPE, id, deny, allow);
+        overrides.add(override);
+        return this;
+    }
+
+    public ChannelAction addPermissionOverride(Member member, Collection<Permission> allow, Collection<Permission> deny)
+    {
+        Args.notNull(member, "Override Member");
+        if (!member.getGuild().equals(guild))
+            throw new IllegalArgumentException("Specified Member is not in the same Guild!");
+
+        String id = member.getUser().getId();
+        PromisePermissionOverride override = new PromisePermissionOverride(USER_TYPE, id, deny, allow);
+        overrides.add(override);
+        return this;
+    }
+
+    // --voice only--
+    public ChannelAction setBitrate(Integer bitrate)
+    {
+        if (bitrate != null && bitrate < 8000)
+            throw new IllegalArgumentException("Bitrate must be greater than 8000.");
+        this.bitrate = bitrate;
+        return this;
+    }
+
+    public ChannelAction setUserlimit(Integer userlimit)
+    {
+        if (userlimit != null && (userlimit < 0 || userlimit > 99))
+            throw new IllegalArgumentException("Userlimit must be between 0-99!");
+        this.userlimit = userlimit;
+        return this;
+    }
+
+    @Override
+    protected void finalizeData()
+    {
+        JSONObject data = new JSONObject();
+        data.put("name", name);
+        data.put("type", voice ? 2 : 0);
+        data.put("permission_overwrites", new JSONArray(overrides));
+        if (voice)
+        {
+            if (bitrate != null)
+                data.put("bitrate", bitrate.intValue());
+            if (userlimit != null)
+                data.put("user_limit", userlimit.intValue());
+        }
+
+        this.data = data;
+        super.finalizeData();
+    }
+
+    @Override
+    protected void handleResponse(Response response, Request request)
+    {
+        if (!response.isOk())
+        {
+            request.onFailure(response);
+            return;
+        }
+
+        EntityBuilder builder = EntityBuilder.get(api);
+        Channel channel = voice
+                ? builder.createVoiceChannel(response.getObject(), guild.getId())
+                : builder.createTextChannel(response.getObject(),  guild.getId());
+
+        request.onSuccess(channel);
+    }
+
+    protected final class PromisePermissionOverride implements JSONString
+    {
+        protected final String id;
+        protected final Set<Permission> deny;
+        protected final Set<Permission> allow;
+        protected final int type;
+
+        public PromisePermissionOverride(int type, String id, Collection<Permission> deny, Collection<Permission> allow)
```

Comment:
Due to needing to provide the `long` overrides discussed above, this should take only `long` for allow and deny, and force the addPermissionOverride in ChannelAction to convert the Collection provided to a long before passing to constructor.

Additionally, keep the order of allow-deny the same between the add override methods and the PromisePermissionOverride constructor (currently opposite of eachother)